Hasitha-Su/Nihon-Ninja
# Feature/add di and local db

# Pull Request Description

## Changes Made
In this pull request, we have made several significant changes to the project:

1. **Quiz Functionality**: We implemented a translation quiz feature in the `TranslationQuizFragment`. This feature allows users to practice translation by providing a sentence and a set of words. Users need to select the words in the correct order to form a sentence.

2. **Navigation Enhancement**: We added navigation between quiz questions and a results screen. After completing the quiz, users are taken to a results screen where they can see their score.

3. **Data Integration**: We integrated quiz data from both a local Room database and a remote API using Retrofit. The data is fetched from the API, stored in the local database, and then used to populate the quiz questions and answers.

4. **ViewModel and Repository**: We introduced `ViewModel` and `Repository` classes to handle the business logic and data retrieval for the quiz feature.

5. **UI Enhancements**: We improved the user interface by styling buttons and text views, and ensured a responsive design that adapts to different screen sizes.

6. **Code Clean-up**: We refactored and organized code, removed unused resources, and optimized the layout.

## How to Test
To test these changes, follow these steps:
1. Launch the app on an emulator or physical device.
2. Navigate to the quiz section.
3. Start a translation quiz and complete it.
4. Observe the navigation between quiz questions and the results screen.
5. Check if quiz data is being fetched and displayed correctly.
